### PR TITLE
python3Packages.zeep: 4.0.0 -> 4.1.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 4.1.0

Change log: https://github.com/mvantellingen/python-zeep/releases/tag/4.1.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Depends on #134674

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 134667` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>lexicon</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>home-assistant</li>
    <li>python38Packages.platformdirs</li>
    <li>python38Packages.total-connect-client</li>
    <li>python38Packages.zeep</li>
    <li>python39Packages.platformdirs</li>
    <li>python39Packages.total-connect-client</li>
    <li>python39Packages.zeep</li>
  </ul>
</details>

File: pkgs/development/python-modules/platformdirs/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  checkInputs = [
+    appdirs
+    pytest-mock
+    pytestCheckHook
+  ];
+
+ pythonImportsCheck = [ "platformdirs" ];
+
+  meta = with lib; {
+    description = "PPython module for determining appropriate platform-specific directories";
```
Comment (risicle) [inline pkgs/development/python-modules/platformdirs/default.nix:39]: > PPython

typo?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 134667` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python38Packages.sunpy: log was empty</li>
<li>python39Packages.sunpy: log was empty</li>
  </ul>
</details>
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>home-assistant</li>
<li>lexicon</li>
<li>python38Packages.platformdirs</li>
<li>python38Packages.total-connect-client</li>
<li>python38Packages.zeep</li>
<li>python39Packages.platformdirs</li>
<li>python39Packages.total-connect-client</li>
<li>python39Packages.zeep</li>
</ul>
</details>

**Comment by risicle** [comment]
`nixpkgs-review` happy macos 10.15 & nixos x86_64 (skipped `homeassistant` though)

File: pkgs/development/python-modules/platformdirs/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/platformdirs/default.nix]: Thanks

**Pull request merged**